api/server/server.go

```diff
@@ -876,16 +876,43 @@ func deleteContainers(eng *engine.Engine, version version.Version, w http.Respon
 	if vars == nil {
 		return fmt.Errorf("Missing parameter")
 	}
-	job := eng.Job("rm", vars["name"])
 
-	job.Setenv("forceRemove", r.Form.Get("force"))
+	name := vars["name"]
+	if name == "" {
+		return fmt.Errorf("Container name cannot be empty")
+	}
 
-	job.Setenv("removeVolume", r.Form.Get("v"))
-	job.Setenv("removeLink", r.Form.Get("link"))
-	if err := job.Run(); err != nil {
+	var forceRemove, removeVolume, removeLink bool
+
+	if force := r.Form.Get("force"); force != "" {
+		fr, err := strconv.ParseBool(force)
```

Comment:
mmm why changing behaviour? jobs arguments were parsed to bool. Ok to remove vars btw